feat: add `ndarray/base/nullary-strided1d-dispatch-factory`

headlessNode opened the pull request:
Progresses #2656.

## Description

> What is the purpose of this pull request?

This pull request:

-   add `ndarray/base/nullary-strided1d-dispatch-factory`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2656

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add ndarray/base/nullary-strided1d-dispatch-factory

headlessNode commented:
Need to merge https://github.com/stdlib-js/stdlib/pull/7821, https://github.com/stdlib-js/stdlib/pull/7772 and https://github.com/stdlib-js/stdlib/pull/7756 first.

kgryte commented:
@headlessNode I did a good bit of refactoring in https://github.com/stdlib-js/stdlib/pull/7821. You'll need to update this PR accordingly.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into ndarray-nullary1d-dispatch-factory

headlessNode committed:
Merge branch 'stdlib-js:develop' into ndarray-nullary1d-dispatch-factory

headlessNode committed:
refactor: apply suggestions from code review

headlessNode committed:
refactor: apply suggestions from code review

headlessNode committed:
fix: lint error

headlessNode committed:
fix: lint error

headlessNode commented:
Lint error expected.

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: skipped
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented:
As the REPL text lint error is expected, I'll go ahead and merge.
